xds/internal/client/xds.go

```diff
@@ -616,6 +616,37 @@ func validateClusterAndConstructClusterUpdate(cluster *v3clusterpb.Cluster) (Clu
 		return ret, nil
 	case cluster.GetType() == v3clusterpb.Cluster_LOGICAL_DNS:
 		ret.ClusterType = ClusterTypeLogicalDNS
+		loadAssignment := cluster.GetLoadAssignment()
```

Comment:
Nit: Can we move this to a helper function like validateLoadAssignment(loadAssignment) error? This feels clunky to have this in the switch, I think it'd be more readable that way.

Reply:
Done